### PR TITLE
Add Packet support 

### LINKED ISSUE #3352: Add Packet support to Kubermatic
**User Story**
As a user, I want to use Kubermatic to create Kubernetes Clusters on Packet

**Acceptance criteria**

It is possible to use Packet as a provider to create Kubermatic-managed clusters on

### PR DESCRIPTION
**Pull request by nikhita**
**What this PR does / why we need it**:

Adds Packet support to Kubermatic.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3352 

**Does this PR introduce a user-facing change?**:

```release-note
It is now possible to create Kubermatic-managed clusters on Packet.
```

/assign @alvaroaleman

### CODE REVIEW

**Comment by nikhita** [comment]
Needs https://github.com/kubermatic/machine-controller/pull/523

**Comment by nikhita** [comment]
The packet e2e test is green! :sparkles: 

This is now ready for review.

File: api/pkg/resources/machine/common.go [hunk 1]
```diff
@@ -224,6 +225,35 @@ func getDigitaloceanProviderSpec(c *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpe
...
 }
 
+func getPacketProviderSpec(c *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc provider.DatacenterMeta) (*runtime.RawExtension, error) {
+	config := packet.RawConfig{
+		InstanceType: providerconfig.ConfigVarString{Value: nodeSpec.Cloud.Packet.InstanceType},
+		OS:           providerconfig.ConfigVarString{Value: nodeSpec.Cloud.Packet.OS},
+	}
+
+	tags := sets.NewString(nodeSpec.Cloud.Packet.Tags...)
+	tags.Insert("kubermatic", "kubermatic-cluster-"+c.Name)
```
Comment (alvaroaleman) [inline api/pkg/resources/machine/common.go]: Can we use the string `kubernetes` instead of `kubermatic` here? Some customers didn't want the latter

**Review by alvaroaleman — COMMENTED** [review]
One nit, other than that lgtm :)

/assign @kubermatic/ui-team

File: api/pkg/resources/machine/common.go (hunk 1 above)
Comment (nikhita) [reply api/pkg/resources/machine/common.go]: Done.

openstack and digitalocean both use `kubermatic` though..should they be updated as well? :thinking:

**Comment by nikhita** [comment]
> /assign @kubermatic/ui-team

There is no such team :grimacing: 

/assign @p0lyn0mial

File: api/pkg/resources/machine/common.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/resources/machine/common.go]: Yes, they should

/lgtm
/approve

**Comment by alvaroaleman** [comment]
> There is no such team :grimacing:

There is, you can click on the link and you will end up here: https://github.com/orgs/kubermatic/teams/ui-team

Seems Prow cant handle assignment to a team thought

**Comment by nikhita** [comment]
> Seems Prow cant handle assignment to a team thought

It's not prow, it's GitHub ;)

GitHub doesn't support team assignments yet.

**Comment by nikhita** [comment]
/test pull-kubermatic-e2e-packet-coreos-1.14

Just want to make sure we run this before merging.

**Pull request merged**